### PR TITLE
paraswap: add paraswap_zkevm + v6.1 spell

### PR DESCRIPTION
**Pull request by mwamedacen**
This PR is a follow up of previous works (copy pasta). Previous PRs: https://github.com/duneanalytics/spellbook/pull/5597 and https://github.com/duneanalytics/spellbook/pull/6002

### CODE REVIEW

File: models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    alias = 'trades',    
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'method', 'trace_address'],
+    post_hook='{{ expose_spells(\'["zkevm"]\',
+                                "project",
+                                "paraswap_v6",
+                                \'["eptighte", "mwamedacen"]\') }}'
```
Comment (Hosuke) [inline models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql:12]: We can apply the new format for post_hook here:
```suggestion
    post_hook='{{ expose_spells(blockchains = \'["zkevm"]\',
                                spell_type = "project",
                                spell_name = "paraswap_v6",
                                contributors = \'["eptighte", "mwamedacen"]\') }}'
```

File: models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            THEN 0x4f9a0e7fd2bf6067db6994cf12e4495df938e6e9 -- WETH 
+            ELSE from_hex(srcToken)
+        END AS token_sold_address,
+        projectContractAddress as project_contract_address,
+        txHash AS tx_hash, 
+        callTraceAddress AS trace_address,
+        CAST(-1 as integer) AS evt_index
+    FROM {{ ref('paraswap_v6_zkevm_trades_decoded') }}     
+    {% if is_incremental() %}
+    WHERE blockTime >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql:44]: We can use the `incremental_predicate` here and for all incremental filters:
```suggestion
    WHERE {{ incremental_predicate('blockTime') }}
```

**Review by Hosuke — CHANGES_REQUESTED** [review]

**Comment by mwamedacen** [comment]
thanks for the suggestions, applied @Hosuke both. Worth noting that those improvements are not normalised across the other spells (not only ParaSwap ones)

File: models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    d.trace_address,
+    d.evt_index
+FROM dexs d
+INNER JOIN {{ source('zkevm', 'transactions') }} tx ON d.tx_hash = tx.hash
+    AND d.block_number = tx.block_number
+    {% if not is_incremental() %}
+    AND tx.block_time >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql:104]:
```suggestion
    AND {{ incremental_predicate('tx.block_time') }}
```

File: models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+LEFT JOIN price_missed_next p_next1 ON d.token_bought_address = p_next1.contract_address
+    AND d.block_time > p_next1.minute -- Swap after last price record time
+LEFT JOIN {{ source('prices', 'usd') }} p2 ON p2.minute = date_trunc('minute', d.block_time)
+    AND p2.contract_address = d.token_sold_address
+    AND p2.blockchain = 'zkevm'
+    {% if not is_incremental() %}
+    AND p2.minute >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND p2.minute >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql:130]:
```suggestion
    AND {{ incremental_predicate('p2.minute') }}
```

File: models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+LEFT JOIN {{ source('tokens', 'erc20') }} e2 on e2.contract_address = d.token_sold_address
+    AND e2.blockchain = 'zkevm'
+LEFT JOIN {{ source('prices', 'usd') }} p1 ON p1.minute = date_trunc('minute', d.block_time)
+    AND p1.contract_address = d.token_bought_address
+    AND p1.blockchain = 'zkevm'
+    {% if not is_incremental() %}
+    AND p1.minute >= TIMESTAMP '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND p1.minute >= date_trunc('day', now() - interval '7' day)
```
Comment (Hosuke) [inline models/paraswap/zkevm/paraswap_v6_zkevm_trades.sql:117]:
```suggestion
    AND {{ incremental_predicate('p1.minute') }}
```

**Review by Hosuke — CHANGES_REQUESTED** [review]
These improvements are new standards that we are striving to implement in our development process. We will do our best to cover these in all new spells. Thank you for your understanding and support!

**Comment by jeff-dude** [comment]
i've been letting this one sit, as it appears suggestion still remain. please do make the changes and look to finalize

**Comment by jeff-dude** [comment]
closing due to inactivity. feel free to reopen and describe what needs to happen next, if necessary. thank you.

**Pull request closed without merging**